From efd5d6325050cb3592d830c02c83046029c23971 Mon Sep 17 00:00:00 2001
From: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Date: Thu, 25 Feb 2021 20:15:08 +0800
Subject: [PATCH] arm64: fix compile error when CONFIG_ACPI is not enabled
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: efd5d6325050cb3592d830c02c83046029c23971

MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

hulk inclusion
category: bugfix
bugzilla: 47994
CVE: NA

-------------------------------------------------------------------------

Fix the following compile error when CONFIG_ACPI is not enabled.

arch/arm64/kernel/smp.c: In function ‘smp_prepare_cpus’:
arch/arm64/kernel/smp.c:785:9: error: ‘cpu_madt_gicc’ undeclared (first use in this function); did you mean ‘bpf_map_inc’?
    if ((cpu_madt_gicc[cpu].flags & ACPI_MADT_ENABLED))
         ^~~~~~~~~~~~~
         bpf_map_inc
arch/arm64/kernel/smp.c:785:9: note: each undeclared identifier is reported only once for each function it appears in
make[3]: *** [arch/arm64/kernel/smp.o] Error 1
make[3]: *** Waiting for unfinished jobs....

Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/kernel/smp.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/kernel/smp.c b/arch/arm64/kernel/smp.c
index 5628891ea93a..70c7634433e1 100644
--- a/arch/arm64/kernel/smp.c
+++ b/arch/arm64/kernel/smp.c
@@ -781,12 +781,16 @@ void __init smp_prepare_cpus(unsigned int max_cpus)
 		if (err)
 			continue;
 
-		if (acpi_disabled) {
-			set_cpu_present(cpu, true);
-		} else {
+#ifdef CONFIG_ACPI
+		if (!acpi_disabled) {
 			if ((cpu_madt_gicc[cpu].flags & ACPI_MADT_ENABLED))
 				set_cpu_present(cpu, true);
+		} else {
+			set_cpu_present(cpu, true);
 		}
+#else
+		set_cpu_present(cpu, true);
+#endif
 
 		numa_store_cpu_info(cpu);
 	}
-- 
2.26.2

